tests/tests_unit/test_rules/test_models/test_restrictions.py

```diff
@@ -0,0 +1,143 @@
+            (
+                "cardinalityConstraint:has.complex_property-123(maxCardinality,5)",
+                "has.complex_property-123",
+                "maxCardinality",
+                5,
+                None,
+            ),
+        ],
+    )
+    def test_parse_basic_cardinality_constraints(
+        self, data, expected_property, expected_constraint, expected_value, expected_on
+    ):
+        result = ConceptPropertyCardinalityConstraint.load(data)
+
+        assert result.property_ == expected_property
+        assert result.constraint == expected_constraint
+        assert result.value == expected_value
+        assert result.on == expected_on
+
+    @pytest.mark.parametrize(
+        "data,expected_on_type",
+        [
+            ("cardinalityConstraint:hasAge(qualifiedCardinality,1,integer)", DataType),
+            ("cardinalityConstraint:hasVehicle(qualifiedCardinality,3,Car)", ConceptEntity),
+        ],
+    )
+    def test_parse_qualified_cardinality_constraints(self, data, expected_on_type):
+        defaults = {"prefix": "ex"} if "Car" in data else {}
+        result = ConceptPropertyCardinalityConstraint.load(data, **defaults)
+
+        assert result.constraint == "qualifiedCardinality"
+        assert isinstance(result.on, expected_on_type)
+
+    def test_parse_invalid_cardinality_format(self):
+        data = "invalidConstraint:hasProperty(minCardinality,1)"
+        with pytest.raises(ValidationError, match="Invalid cardinality constraint format"):
+            ConceptPropertyCardinalityConstraint.load(data)
+
+
+class TestParseRestriction:
+    @pytest.mark.parametrize(
+        "data,expected_type,expected_property",
+        [
+            ("valueConstraint:hasOwner(hasValue,ni:John)", ConceptPropertyValueConstraint, "hasOwner"),
+            ("cardinalityConstraint:hasChild(minCardinality,2)", ConceptPropertyCardinalityConstraint, "hasChild"),
+            ("valueConstraint:hasType(allValuesFrom,Vehicle)", ConceptPropertyValueConstraint, "hasType"),
+        ],
+    )
+    def test_parse_valid_restrictions(self, data, expected_type, expected_property):
+        defaults = {"prefix": "test"} if "Vehicle" in data else {}
+        result = parse_restriction(data, **defaults)
+
+        assert isinstance(result, expected_type)
+        assert result.property_ == expected_property
+
+    @pytest.mark.parametrize(
+        "invalid_data",
+        [
+            "invalidRestriction:hasProperty(someConstraint,value)",
+            "",
+        ],
+    )
+    def test_parse_invalid_restrictions(self, invalid_data):
+        with pytest.raises(NeatValueError, match="Invalid restriction format"):
+            parse_restriction(invalid_data)
+
+
+class TestNamedIndividualEntity:
+    def test_reset_prefix(self):
+        entity = NamedIndividualEntity(suffix="test", prefix="one-that-gets-reset")
+        entity.reset_prefix()
```

Comment:
```suggestion
```

It is uncommon to call model_validator methods explicitly. They are invoked automatically. You can remove it without the test failing.